src/settings_internal.h

```diff
@@ -164,6 +164,12 @@ struct IntSettingDesc : SettingDesc {
 	 * @param The new value for the setting.
 	 */
 	typedef void PostChangeCallback(int32_t value);
+	/**
+	 * A callback to set the correct default value, in case it can be measured in time
+	 * units or expressed as a percentage.
+	 * @param value The prospective new value for the setting.
+	 */
+	typedef void GetDefaultValueCallback(int32_t &value);
```

Comment:
I would expect a function called `Get...` to return something. What's the rationale behind using a reference, instead of just returning the correct default given the current state of the (other) settings?
The only reason I can think of to use a reference is to not set the default in some cases, but that kind of defeats the purpose. Especially as it was just set to something before from the default value. Why not completely ignore the static default value and only use the result of this callback?

So, around line 2733 of settings_gui.cpp you'd get:
```
		} else if (sd->get_def_cb != nullptr) {
			value = sd->get_def_cb();
		} else {
			value = sd->def;
		}
```

Reply:
Indeed, it makes more sense to just return the correct default value. I've just pushed an update to correct that.